ArduPlane/altitude.cpp

```diff
@@ -24,70 +24,46 @@
   adjust altitude target depending on mode
  */
 void Plane::adjust_altitude_target()
+{
+    control_mode->update_target_altitude();
+}
+
+// to be moved to mode.cpp in a future PR, kept here to minimise diff
+void Mode::update_target_altitude()
 {
     Location target_location;
 
-    if (control_mode == &mode_fbwb ||
-        control_mode == &mode_cruise) {
-        return;
-    }
-    if ((control_mode == &mode_loiter) && plane.stick_mixing_enabled() && (plane.g2.flight_options & FlightOptions::ENABLE_LOITER_ALT_CONTROL)) {
-       return;
-    }
-#if OFFBOARD_GUIDED == ENABLED
-    if (control_mode == &mode_guided && ((guided_state.target_alt_time_ms != 0) || guided_state.target_alt > -0.001 )) { // target_alt now defaults to -1, and _time_ms defaults to zero.
-        // offboard altitude demanded
-        uint32_t now = AP_HAL::millis();
-        float delta = 1e-3f * (now - guided_state.target_alt_time_ms);
-        guided_state.target_alt_time_ms = now;
-        // determine delta accurately as a float
-        float delta_amt_f = delta * guided_state.target_alt_accel;
-        // then scale x100 to match last_target_alt and convert to a signed int32_t as it may be negative
-        int32_t delta_amt_i = (int32_t)(100.0 * delta_amt_f); 
-        Location temp {};
-        temp.alt = guided_state.last_target_alt + delta_amt_i; // ...to avoid floats here, 
-        if (is_positive(guided_state.target_alt_accel)) {
-            temp.alt = MIN(guided_state.target_alt, temp.alt);
-        } else {
-            temp.alt = MAX(guided_state.target_alt, temp.alt);
-        }
-        guided_state.last_target_alt = temp.alt;
-        set_target_altitude_location(temp);
-    } else 
-#endif // OFFBOARD_GUIDED == ENABLED
-      if (control_mode->update_target_altitude()) {
-          // handled in mode specific code
-    } else if (landing.is_flaring()) {
+    if (plane.landing.is_flaring()) {
         // during a landing flare, use TECS_LAND_SINK as a target sink
         // rate, and ignores the target altitude
-        set_target_altitude_location(next_WP_loc);
-    } else if (landing.is_on_approach()) {
-        landing.setup_landing_glide_slope(prev_WP_loc, next_WP_loc, current_loc, target_altitude.offset_cm);
-        landing.adjust_landing_slope_for_rangefinder_bump(rangefinder_state, prev_WP_loc, next_WP_loc, current_loc, auto_state.wp_distance, target_altitude.offset_cm);
-    } else if (landing.get_target_altitude_location(target_location)) {
-       set_target_altitude_location(target_location);
+        plane.set_target_altitude_location(plane.next_WP_loc);
```

Comment:
can we have the adding of the plane. as a separate commit? that should make the diff a lot more readable